tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -396,6 +440,308 @@ def test_python_lambda_subscribe_sns_topic(
+                EventSourceArn=stream_arn,
+                MaximumBatchingWindowInSeconds=1,
+                MaximumRetryAttempts=1,
+                DestinationConfig=destination_config,
+            )
+
+            event_source_mapping_uuid = result["UUID"]
+            event_source_mapping_state = result["State"]
+            if event_source_mapping_state != "Enabled":
+
+                def check_mapping_state():
+                    state = lambda_client.get_event_source_mapping(UUID=event_source_mapping_uuid)[
+                        "State"
+                    ]
+                    if state != "Enabled":
+                        raise Exception(f"State is {state}")
+
+                retry(check_mapping_state, retries=6, sleep_before=2.0, sleep=1.0)
+
+            insert = {partition_key: {"S": "hello world"}}
+
+            dynamodb_client.put_item(TableName=table_name, Item=insert)
+
+            def verify_failure_received():
+                res = sqs_client.receive_message(QueueUrl=queue_event_source_mapping)
+                msg = res["Messages"][0]
+                body = json.loads(msg["Body"])
+                assert body["requestContext"]["condition"] == "RetryAttemptsExhausted"
+                assert body["DDBStreamBatchInfo"]["batchSize"] == 1
+                assert body["DDBStreamBatchInfo"]["streamArn"] in stream_arn
+
+            retry(verify_failure_received, retries=50, sleep=5, sleep_before=5)
+
+        finally:
+            lambda_client.delete_event_source_mapping(UUID=event_source_mapping_uuid)
+
+    def test_event_source_mapping_dynamodb(
+        self,
+        lambda_client,
+        create_lambda_function,
+        iam_create_role_with_policy,
+        dynamodb_client,
+        dynamodb_create_table,
+        logs_client,
+        check_lambda_logs,
+    ):
+        try:
+            function_name = f"lambda_func-{short_uid()}"
+            role = f"test-lambda-role-{short_uid()}"
+            policy_name = f"test-lambda-policy-{short_uid()}"
+            role_arn = iam_create_role_with_policy(
+                RoleName=role,
+                PolicyName=policy_name,
+                RoleDefinition=lambda_role,
+                PolicyDefinition=s3_lambda_permission,
+            )
+
+            create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON_ECHO,
+                func_name=function_name,
+                runtime=LAMBDA_RUNTIME_PYTHON37,
+                role=role_arn,
+                libs=TEST_LAMBDA_LIBS,
+            )
+
+            # test destination-config for create_event_source_mapping
+            table_name = f"test-table-{short_uid()}"
+
+            partition_key = "my_partition_key"
+            result = dynamodb_create_table(table_name=table_name, partition_key=partition_key)
+
+            status = result["TableDescription"]["TableStatus"]
+            # wait for status "ACTIVE"
+            if status != "ACTIVE":
+
+                def check_table_active():
```

Comment:
Again, if you move this to its own scope you can reuse of it in your other tests 👍